### DISCUSSION TITLE
Driver enable issue

### DISCUSSION THREAD

**Discussion by LunchM** [openTCS/opentcs#66] [Q&A]
I wrote the `enable()` function in the `VehicleCommAdapter` class as following:
```
@Override
  public synchronized void enable() {
    try{
      String strg = "try to enable\n";
      File file = new File("d:/function_is_called.txt");
      if(!file.exists()){
        file.createNewFile();
      }
      FileWriter fileWriter = new FileWriter(file.getName(), true);
      fileWriter.write(strg);
      fileWriter.close();
    }catch (IOException e){
      e.printStackTrace();
    }
    System.out.println("try to enable\n");
    if (isEnabled()) {
      return;
    }
    getProcessModel().getVelocityController().addVelocityListener(getProcessModel());
    connectVehicle();
    super.enable();
  }
```
And it happened that when I tick the `enable` box in the KernelControlCenter GUI, nothing happened. It seemed that the `enable()` function was not called. I wonder what happened and how to fix this.

**Comment by swltr** [comment]
If `enable()` wasn't called, your vehicle driver may not have been registered with the kernel and attached to the respective vehicle.

**Reply by LunchM** [reply]
I directly changed the code in `LoopbackCommunicationAdapter` as above so that I can make sure it is attached to the respective vehicle in the KernelControlCenter. However, the same problem remains.

**Reply by swltr** [reply]
You have not explicitly stated why you think `enable()` is not being called. I assume you do not see the file being created and/or the output of `println()` being shown. If that is the case, I suspect that the kernel is still running the unmodified driver, as merely adding a log statement to `enable()` and enabling the driver in the KCC does show the log output for me.

Which version of openTCS is this, and how exactly do you build and run the kernel with the modified driver?

**Reply by LunchM** [reply]
I'm using openTCS 5.6.
I used IntelliJ IDEA and built the project with gradle. Then I enter `opentcs/build/install/openTCS` and run `openTCS-Kernel/startKernel.bat` and `openTCS-KernelControlCenter/startKernelControlCenter.bat`. I loaded `Demo-01.xml` and uploaded it to the kernel. And then I tried to enable the driver in KernelControlCenter and no output shown.

**Reply by swltr** [reply]
Do you see the output when you use the logger instead of `System.out.println()`? Does it also not create the file?

What happens if you throw an exception or add a call to `System.exit()` in there?
